### PR TITLE
Migrate patch building tests

### PR DESCRIPTION
**Pull request by jesseduffield**
**PR Description**

Migrates the patch building tests

- **Please check if the PR fulfills these requirements**

* [x] Cheatsheets are up-to-date (run `go run scripts/cheatsheet/main.go generate`)
* [x] Code has been formatted (see [here](https://github.com/jesseduffield/lazygit/blob/master/CONTRIBUTING.md#code-formatting))
* [x] Tests have been added/updated (see [here](https://github.com/jesseduffield/lazygit/blob/master/pkg/integration/README.md) for the integration test guide)
* [x] Text is internationalised (see [here](https://github.com/jesseduffield/lazygit/blob/master/CONTRIBUTING.md#internationalisation))
* [x] Docs (specifically `docs/Config.md`) have been updated if necessary
* [x] You've read through your own file changes for silly mistakes etc

### CODE REVIEW

File: pkg/integration/components/assertion_helper.go [hunk 1]
```diff
@@ -11,9 +11,9 @@ type assertionHelper struct {
 }
 
 // milliseconds we'll wait when an assertion fails.
-var retryWaitTimes = []int{0, 1, 1, 1, 1, 1, 5, 10, 20, 40, 100, 200, 500, 1000, 2000, 4000}
+var retryWaitTimes = []int{0, 1, 1, 1, 1, 1, 5, 10, 20, 40, 100}
```
Comment (stefanhaller) [inline pkg/integration/components/assertion_helper.go:14]: Thanks for this! 😄  (I was always wondering why we have to wait so long.)

Comment (jesseduffield) [reply pkg/integration/components/assertion_helper.go:14]: Ah actually that was accidentally committed: I was reducing the timeout for my own local usage. But if CI can handle it then I'm fine to keep it that way across the board

File: pkg/integration/components/test_driver.go [hunk 2]
```diff
@@ -71,65 +70,6 @@ func (self *TestDriver) Shell() *Shell {
 	return self.shell
 }
 
-// this will look for a list item in the current panel and if it finds it, it will
```
Comment (jesseduffield) [inline pkg/integration/components/test_driver.go:74]: moving to view_driver.go

File: pkg/integration/components/assertion_helper.go (hunk 1 above)
Comment (stefanhaller) [reply pkg/integration/components/assertion_helper.go:14]: Ah. 😄 

Maybe we could have an environment variable that is only set on CI and that makes it wait longer.

File: pkg/integration/tests/patch_building/start_new_patch.go [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+		t.Views().CommitFiles().
+			IsFocused().
+			Lines(
+				Contains("file1").IsSelected(),
+			).
+			PressPrimaryAction().
+			Tap(func() {
+				t.ExpectPopup().Confirmation().
+					Title(Contains("")).
+					Content(Contains("")).
```
Comment (jesseduffield) [inline pkg/integration/tests/patch_building/start_new_patch.go:56]: whoops forgot to populate these

**Pull request merged**